### ISSUE TITLE
Pipenv does not support installing from local source packages

### ISSUE THREAD

**Issue by chrish42** [kennethreitz/pipenv#817]
I created a virtualenv with pipenv, and installed some required packages from PyPI with it, which works well. However, I also have in my project a Python source distribution tarball (created by running `python setup.py sdist` in another project), that I'd like to install in the virtualenv. When running `pipenv install mytarball.tar.gz`, the source tarball is successfully installed in the virtualenv, but then I get an error at the end of the process:

```
Adding tarball-1.5.2.tar.gz to Pipfile's [packages]…
Locking [dev-packages] dependencies…
Locking [packages] dependencies…
Traceback (most recent call last):
  File "/Users/christianhudon/miniconda3/bin/pipenv", line 11, in <module>
    sys.exit(cli())
  File "/Users/christianhudon/miniconda3/lib/python3.6/site-packages/pipenv/vendor/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/Users/christianhudon/miniconda3/lib/python3.6/site-packages/pipenv/vendor/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/Users/christianhudon/miniconda3/lib/python3.6/site-packages/pipenv/vendor/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/christianhudon/miniconda3/lib/python3.6/site-packages/pipenv/vendor/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/christianhudon/miniconda3/lib/python3.6/site-packages/pipenv/vendor/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/Users/christianhudon/miniconda3/lib/python3.6/site-packages/pipenv/cli.py", line 1852, in install
    do_lock(system=system, pre=pre)
  File "/Users/christianhudon/miniconda3/lib/python3.6/site-packages/pipenv/cli.py", line 1083, in do_lock
    pre=pre
  File "/Users/christianhudon/miniconda3/lib/python3.6/site-packages/pipenv/utils.py", line 421, in resolve_deps
    resolved_tree.update(resolver.resolve(max_rounds=PIPENV_MAX_ROUNDS))
  File "/Users/christianhudon/miniconda3/lib/python3.6/site-packages/pipenv/patched/piptools/resolver.py", line 88, in resolve
    self.their_constraints))
  File "/Users/christianhudon/miniconda3/lib/python3.6/site-packages/pipenv/patched/piptools/resolver.py", line 125, in check_constraints
    raise UnsupportedConstraint(msg, constraint)
piptools.exceptions.UnsupportedConstraint: pip-compile does not support URLs as packages, unless they are editable. Perhaps add -e option? (constraint was: file:///Users/christianhudon/test/tarball-1.5.2.tar.gz (from -r /var/folders/t0/_kvmbqhs4n92fg18mvdyd12w0000gq/T/pipenv-9i1ndzrb-requirement.txt (line 1)))
```

**Comment by techalchemy** [comment]
I think this is related to #806 and can be fixed with a similar approach to #807

**Comment by techalchemy** [comment]
This is now properly fixed in #958
